core/modules/query/Constraint.h

```diff
@@ -45,6 +45,12 @@ namespace query {
 /// A detected qserv constraint for C++ to Python
 class Constraint {
 public:
+    Constraint() = default;
+
+    Constraint(std::string const& name_, std::vector<std::string> const& params_)
+            : name(name_), params(params_) {
+    }
+
     std::string name;
     std::vector<std::string> params;
```

Comment:
If `name` and `params` are only set by the constructor(s) then they may be declared as `const`